### ISSUE TITLE
[java] Improve BadComparison and rename to ComparisonWithNaN

### ISSUE THREAD

**Issue by oowekyala** [pmd/pmd#2895]
**Is your feature request related to a problem? Please describe.**
* The documentation of [errorprone.xml/BadComparison](https://pmd.github.io/pmd-6.29.0/pmd_rules_java_errorprone.html#badcomparison) is wrong. The problem with comparing some number to `Double.NaN` is not floating-point precision, it's that such a comparison always returns false. Double and Float.NaN are not even equal to themselves:
```java
jshell> Double.NaN == Double.NaN
$1 ==> false
jshell> Double.NaN <= Double.NaN
$2 ==> false
```

* As evidenced by the test above, the rule should care about comparison operators other than `==` and `!=` 
* The rule name is really vague. Consider renaming it to `ComparisonWithNaN`

**Describe the solution you'd like** Described above
